### PR TITLE
add authority parameter to consensus params

### PR DESCRIPTION
**Pull request by vladjdk**
partially closes: https://linear.app/cosmoslabs/issue/STACK-1856/replace-authority-keeper-params-with-consensus-parameters-authority

---

> [!NOTE]
> Introduces `AuthorityParams` to consensus params with default, validation (<=256 chars), full proto/plumbing, and tests, plus changelog update.
> 
> - **Types**:
>   - Add `AuthorityParams` and `ConsensusParams.Authority`; provide `DefaultAuthorityParams` (hash of "gov"), `ValidateBasic` max length 256, `Update`, `ToProto`/`FromProto` wiring.
>   - Extend tests to cover authority validation, updates, and proto round-trip; adjust helpers to accept `authority`.
> - **Proto**:
>   - Introduce `AuthorityParams` message and add `authority` field to `ConsensusParams`; regenerate Go code (getters, marshal/unmarshal, size, equal).
> - **Docs**:
>   - Update `CHANGELOG.md` with validation note for `AuthorityParams.Authority`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6774f2a3c798fb8274e60625d08a1e28ab32bec8. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: types/params.go [hunk 1]
```diff
@@ -139,6 +147,12 @@ func DefaultABCIParams() ABCIParams {
 	}
 }
 
+func DefaultAuthorityParams() AuthorityParams {
+	return AuthorityParams{
+		Authority: crypto.AddressHash([]byte("gov")).String(),
```
Comment (swift1337) [inline types/params.go]: I don't think that it's backward compatible as an existing chain would have `""`

File: types/params.go [hunk 2]
```diff
@@ -374,5 +401,8 @@ func ConsensusParamsFromProto(pbParams cmtproto.ConsensusParams) ConsensusParams
 	if pbParams.Abci != nil {
 		c.ABCI.VoteExtensionsEnableHeight = pbParams.Abci.GetVoteExtensionsEnableHeight()
 	}
+	if pbParams.Authority != nil {
+		c.Authority.Authority = pbParams.Authority.Authority
+	}
```
[the remark this answers is not in the document]
Comment (swift1337) [reply types/params.go:410]: agree

File: types/params_test.go [hunk 3]
```diff
@@ -230,14 +253,17 @@ func TestConsensusParamsUpdate_VoteExtensionsEnableHeight(t *testing.T) {
...
+		makeParams(1, 4, 3, 1, valEd25519, 1, ""),
+		makeParams(1, 2, 4, 1, valEd25519, 1, ""),
+		makeParams(2, 5, 7, 1, valEd25519, 1, ""),
+		makeParams(1, 7, 6, 1, valEd25519, 1, ""),
+		makeParams(9, 5, 4, 1, valEd25519, 1, ""),
+		makeParams(7, 8, 9, 1, valEd25519, 1, ""),
+		makeParams(4, 6, 5, 1, valEd25519, 1, ""),
+		makeParams(1, 2, 3, 1, valEd25519, 1, ""),
+		makeParams(1, 2, 3, 1, valEd25519, 1, "governance-module"),
+		makeParams(1, 2, 3, 1, valEd25519, 1, "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn"),
```
Comment (swift1337) [inline types/params_test.go:262]: Can you add a test case that showcases backward (when Authority is nil)?

File: types/params.go (hunk 1 above)
Comment (vladjdk) [reply types/params.go]: It's not backward-compatible. Each chain would need to specifically set an authority during an upgrade. The default is "", which means that there is no authority on the modules.

Since Comet has no knowledge of bech32 addresses, I removed this default completely and we defer to the app-side for validation.

File: proto/tendermint/types/params.proto [hunk 4]
```diff
@@ -90,3 +91,7 @@ message ABCIParams {
   // to the application to use when proposing a block during PrepareProposal.
   int64 vote_extensions_enable_height = 1;
 }
+
+message AuthorityParams {
```
Comment (aljo242) [inline proto/tendermint/types/params.proto:99]: needs doc

File: types/params_test.go (hunk 3 above)
Comment (aljo242) [reply types/params_test.go:262]: bump

Comment (vladjdk) [reply types/params_test.go:262]: I changed the default to "", which means that when an application upgrades the SDK version to use this parameter, the upgrade is done there, where the application sets a value.

This is also why I can't add a test proving backwards-compatibility—this isn't backwards-compatible and chains need to set the value via UpgradeHandler. This will be handled by https://github.com/cosmos/cosmos-sdk/pull/25607 when it's done.

File: types/params.go (hunk 1 above)
Comment (vladjdk) [reply types/params.go]: Changed the default to "" to make it more clear that this is deferred to the app-side

File: proto/tendermint/types/params.proto (hunk 4 above)
Comment (vladjdk) [reply proto/tendermint/types/params.proto:99]: added
